theme-app-extension/blocks/embed.liquid

```diff
@@ -0,0 +1,41 @@
+{% assign settings = app.metafields.app_settings.general_setting %}
+
+{% assign quriobot_paths = settings.value.quriobot_path | strip | newline_to_br | strip_newlines | split: "<br />" %}
+
+{% assign quriobot_init = settings.value.quriobot_init %}
+
+<script type="text/javascript">
+  if (!Array.isArray(window.qbOptions)) {
+    window.qbOptions = []
+  }
+  window.qbOptions.push({ "baseUrl": "https://botsrv2.com", "use": "{{ quriobot_paths.first }}" });
+</script>
+<script
+  type="text/javascript"
```

Comment:
this doesn't use the data pulled from app data as agreed

Reply:
Yes. Like I said in the latest email, within this file, we can only read app data but not able to update it when expried

Reply:
We can approach it in this way:

- The Shopify app will have an API that call to Quirobot API to get embed code for specific bot path
- The Shopify will also save it to database so it only need call Quirobot API again when the last embed code is expired.
- Everytime people visit the store, the script will call to our app's API to get embed code then render it

What do you think ?

Reply:
Ok, then how about this approach:
In the admin part, in the app settings, when you save them, use the embed call to pull the code and save it to the app settings. (also have a notice there that in order to update the embedding to the latest version, the user needs to save, or have a separate button for it)
Then in this template, use the app data's embed code.

Reply:
Done it. Let's check